test/testAdaptors/AuraERC4626Adaptor.t.sol

```diff
@@ -0,0 +1,292 @@
+
+/**
+ * @title AuraERC4626AdaptorTest
+ * @author crispymangoes, 0xEinCodes
+ * @notice Cellar Adaptor tests with Aura BPT Pools
+ * @dev Mock datafeeds to be used for underlying BPTs. For tests, we'll go with rETH / wETH BPT pair. We'll use mock datafeeds for the constituent assets of this pair so we can warp forward to simulate reward accrual.
+ * NOTE: transferrance of aura-wrapped BPT is not alowed as per their contracts
+ * TODO: review with Crispy comments outlined within test code
+ * TODO: test with other AuraPools perhaps?
+ * TODO: carry out stateful fuzzing?
+ */
+contract AuraERC4626AdaptorTest is MainnetStarterTest, AdaptorHelperFunctions {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+    using stdStorage for StdStorage;
+
+    AuraERC4626Adaptor private auraERC4626Adaptor;
+
+    ERC4626 public auraRETHWETHBPTVault = ERC4626(address(aura_rETH_wETH_BPT));
+    Cellar private cellar;
+
+    // Chainlink PriceFeeds
+    MockDataFeed private mockWethUsd;
+    MockDataFeed private mockRethUsd;
+    BalancerStablePoolExtension private balancerStablePoolExtension;
+
+    uint32 private wethPosition = 1;
+    uint32 private rethPosition = 2;
+    uint32 private rETH_wETH_BPT_Position = 3;
+    uint32 private auraRethWethBptPoolPosition = 4;
+    uint32 private balPosition = 5;
+    uint32 private auraPosition = 6;
+    uint32 private auraExtras_RETH_WETH_BPT_Position = 7;
+
+    uint256 public initialAssets;
+
+    function setUp() external {
+        // Setup forked environment.
+        string memory rpcKey = "MAINNET_RPC_URL";
+        uint256 blockNumber = 18172424;
+        _startFork(rpcKey, blockNumber);
+
+        // Run Starter setUp code.
+        _setUp();
+
+        auraERC4626Adaptor = new AuraERC4626Adaptor();
+
+        balancerStablePoolExtension = new BalancerStablePoolExtension(priceRouter, IVault(vault));
+        PriceRouter.ChainlinkDerivativeStorage memory stor;
+        PriceRouter.AssetSettings memory settings;
+
+        mockWethUsd = new MockDataFeed(WETH_USD_FEED);
+        mockRethUsd = new MockDataFeed(RETH_ETH_FEED);
+
+        uint256 price = uint256(IChainlinkAggregator(address(mockWethUsd)).latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockWethUsd));
+        priceRouter.addAsset(WETH, settings, abi.encode(stor), price);
+
+        price = uint256(IChainlinkAggregator(address(mockRethUsd)).latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockRethUsd));
+        priceRouter.addAsset(rETH, settings, abi.encode(stor), price);
+
+        price = uint256(IChainlinkAggregator(BAL_USD_FEED).latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, BAL_USD_FEED);
+        priceRouter.addAsset(BAL, settings, abi.encode(stor), price);
+
+        // TODO: AURA doesn't have an AURA_USD Chainlink Feed. For now, we'll make it a mock price feed with WETH FEED
```

Comment:
But we shouldn't need to price Aura, since strategists can just sell it using a DEC Aggregator adaptor

Reply:
Typo meant to type "DEX" like decentralized exchange

Reply:
Awesome, thanks for the clarification. I've removed it.